pyrs/core/summary_generator_stress.py

```diff
@@ -158,13 +156,15 @@ def _write_field_3d(row: int, field: str):
                         strain_field = strain.field
                         strain_value = strain_field.values[row]
                         strain_error = strain_field.errors[row]
-                        entries += _write_number(strain_value) + _write_number(strain_error)
+                        entries += self._write_number(strain_value, decimals) + \
+                            self._write_number(strain_error, decimals)
 
                     elif field == 'Stress':
                         self._stress.select(direction)
                         stress_value = self._stress.values[row]
                         stress_error = self._stress.errors[row]
-                        entries += _write_number(stress_value) + _write_number(stress_error)
+                        entries += self._write_number(stress_value, decimals) + \
+                            self._write_number(stress_error, decimals)
 
                     else:
                         peak_collection = self._get_peak_collection(direction)
```

Comment:
This is confusing. Fit parameters such as `FHWM` are associated with single runs, not with directions. I would understand looping for each direction _and then_ looping for each single run contained in the direction to fish out `Center`, `FWHM`, `Peak_Height`. Something like this:
```
for direction in ('11', '22', '33):
    facade.selection = direction
    facade.stress  # fish out the stress scalar field sample for this direction
    facade.strain  # fish out the strain scalar field sample for this direction
    for run in facace.runs(direction):
        facade.selection = run
        facade.strain  # fish out the strain scalar field sample for this single run
        for param in ('Center', 'Height', 'FWHM'):
            facade.peak_parameter(param)  # fish out the ScalarFieldSample for this peak parameter
```

Plus, using `PeakCollections` will not work in the general case when you have different sample points along different directions, or when you have more than one `PeakCollection` object per direction. The previous code snippet can support these cases

Reply:
@jmborr thanks, would you mind discussing this in the spec issue #514 . Also I thought the facade points were already fused, unless I'm missing something. Thanks.

Reply:
I think it's better if we discuss this on Monday meeting, and after the meeting one-on-one in order to clarify things, but what I can tell you is that you shouldn't be using `PeakCollection` objects to retrieve peak parameters. It only works for a subset of the use-cases.

Reply:
Also, you can take a look to file `test_stress_facade.py`, which contains now more tests, for the use of the facade object

Reply:
OK, thanks for clarifying. I believe this is the summary csv case which predates the facade, I'll accommodate. Let's discuss on Monday.